riscv/csrs.cc

```diff
@@ -1400,3 +1412,39 @@ void henvcfg_csr_t::verify_permissions(insn_t insn, bool write) const {
       throw trap_illegal_instruction(insn.bits());
   }
 }
+
+bool henvcfg_csr_t::unlogged_write(const reg_t val) noexcept {
+  // mask out read-only 0 bits if menvcfg has them turned off
+  return masked_csr_t::unlogged_write((menvcfg->read() | ~(MENVCFG_PBMTE | MENVCFG_STCE)) & val);
+}
+
+stimecmp_csr_t::stimecmp_csr_t(processor_t* const proc, const reg_t addr, const reg_t imask):
+  basic_csr_t(proc, addr, 0), intr_mask(imask) {
+}
+
+bool stimecmp_csr_t::unlogged_write(const reg_t val) noexcept {
+  state->mip->backdoor_write_with_mask(intr_mask, 0);
+  // time will return time + htimedelta if v = true
+  if (state->time->read() >= val) {
```

Comment:
What if `val==0`? This is special-cased in `time_counter_csr_t::sync()` so why not here?

Reply:
0 is technically not a special value and there's no architectural reset value for xtimecmp - I just didn't want the interrupts to fire spuriously at boot time. I can make the reset value -1 or have the interrupts assert the way CLINT is implemented.

Reply:
In `time_counter_csr_t::sync()` you will never turn on STIP if `stimecmp==0`. But here, you will. Why is that behavior different?

It's okay to have interrupts firing at boot time, and MTIP already does (iirc), because `mstatus.MIE=0` will prevent any trap from being taken.